src/main/java/bio/terra/service/dataset/flight/create/CreateDatasetAssetStep.java

```diff
@@ -0,0 +1,35 @@
+package bio.terra.service.dataset.flight.create;
+
+import bio.terra.service.dataset.AssetDao;
+import bio.terra.service.dataset.AssetSpecification;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+
+import java.util.UUID;
+
+public class CreateDatasetAssetStep implements Step {
+
+    private UUID datasetId;
+    private AssetDao assetDao;
+    private AssetSpecification assetSpecification;
+
+    public CreateDatasetAssetStep(UUID datasetId, AssetDao assetDao, AssetSpecification assetSpecification) {
+        this.datasetId = datasetId;
+        this.assetDao = assetDao;
+        this.assetSpecification = assetSpecification;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) {
+        assetDao.create(assetSpecification, datasetId);
+        return StepResult.getStepResultSuccess();
+    }
+
+    @Override
+    public StepResult undoStep(FlightContext context) {
+        assetDao.delete(assetSpecification.getId());
```

Comment:
This won't work. The assetSpecification object in this object is for the to-be-created asset. It doesn't have an id set and it won't get one.

I think what you need to do is this...
In the undoStep:
1. Use the dataset id to fetch the Dataset object
2. Search the Asset list in the dataset object to see if the asset you were trying to create got created. I think you have to do this by name. This only works if we are sure asset names are unique.
3. If the asset is not found, then you are done. It never got created.
4. If the asset is found, then you get its id and call delete.

You cannot assume that the flight object created when the doStep was run is the same flight object when the undoStep is run.